include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_radix_sort_one_wg.h

```diff
@@ -158,7 +158,12 @@ struct __subgroup_radix_sort
                 auto __counter_lacc = __buf_count.get_acc(__cgh);
 
                 __cgh.parallel_for<_Name...>(
-                    __range, ([=](sycl::nd_item<1> __it)[[_ONEDPL_SYCL_REQD_SUB_GROUP_SIZE(__req_sub_group_size)]] {
+                    __range, ([=](sycl::nd_item<1> __it)
+                        // CUDA devices only support sg size of 32, so we disable this requirement
+#if !_ONEDPL_DETECT_NVPTX_COMPILATION
+                        [[_ONEDPL_SYCL_REQD_SUB_GROUP_SIZE(__req_sub_group_size)]]
+#endif
+                    {
```

Comment:
Would it be better to add a macro which folds this logic in?

Something like this:

`__range, ([=](sycl::nd_item<1> __it) _ONEDPL_SYCL_REQD_SUB_GROUP_SIZE_IF_SUPPORTED(__req_sub_group_size) {`

And then keep the logic to check specific hardware in `sycl_defs.h`.

Reply:
Yes, I will add this change.

Reply:
It looks like its valid to have an empty attribute `[[ ]]`, so we can keep the syntax close to the baseline SYCL, and it can look like this:
`__range, ([=](sycl::nd_item<1> __it) [[_ONEDPL_SYCL_REQD_SUB_GROUP_SIZE_IF_SUPPORTED]] (__req_sub_group_size) {`